LISA/Filesystem.cpp

```diff
@@ -178,6 +179,27 @@ long getFreeSpace(const std::string& path)
     return freeSpace;
 }
 
+long getDirectorySpace(const std::string& path)
+{
+    long space{};
+    namespace bf = boost::filesystem;
+    try {
+        if(directoryExists(path)) {
+            for(bf::recursive_directory_iterator it(path); it != bf::recursive_directory_iterator(); ++it)
+            {
+                if(!bf::is_directory(*it)) {
```

Comment:
I had a chance yesterday to do a quick test with my generate bundle: this code crashed on symlinks in the rootfs. I would suggest ignoring symlinks when calculating the used space because they link to other files that take the real space. In my case , symlink did not even resolve correctly because it is meant to be used inside a container: the link pointed to /usr/lib/libxxx, which does not exist inside host rootfs, only inside mounted container rootfs.

Reply:
Yes, I just tested this with symlinks and had a crash when symlink was pointing outside of container. I added symlink skipping.

Reply:
Your check on symlinks does not seem to prevent my crash. I still see a "file not found" crash when those symlinks to link to files that are not there. (they are only there when mounted inside container).
This works for me:
`if(bf::exists(*it) && !bf::is_directory(*it) && !bf::is_symlink(*it)) {`

Also, I should note that hard linked files should not be counted when calculated "used space". Like the files "/usr/lib/dri/*" inside the tarball bundle I shared. Try using the cmdline tool "du" on that dir and you will see what I mean.